### PR TITLE
NFC: 5.9 Documentation update.

### PR DESCRIPTION
**Pull request by zkiraly**
Review and update the documentation for SwiftPM 5.9

### Motivation:

A couple of new API were added to `PackageDescription`, and these needed new documentation, and in some cases review.

### Modifications:

Documentation was added or updated. Nearly all the changes are for post 5.7 API.

### Result:

There will be up-to-date documentation in `PackageDescription` that conforms to the current documentation style.

### CODE REVIEW

**Comment by zkiraly** [comment]
This is ready for engineering review.

**Comment by zkiraly** [comment]
@swift-ci please smoke test.

File: Sources/PackageDescription/Resource.swift [hunk 1]
```diff
@@ -47,7 +47,7 @@ public struct Resource {
 
     /// The rule for the resource.
     let rule: String
-
+
```
Comment (MaxDesiatov) [inline Sources/PackageDescription/Resource.swift:50]: nit: trailing whitespace

File: Sources/PackageDescription/BuildSettings.swift [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 /// The build configuration such as debug or release.
 public struct BuildConfiguration {
+    /// The configuration of the build. Valid values are `debug` and `release`.
     let config: String
```
Comment (neonichu) [inline Sources/PackageDescription/BuildSettings.swift:16]: This should not be public, right? Will it still show up in the developer docs?

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test

**Pull request merged**